athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/PropertyGraphHandler.java

```diff
@@ -105,7 +111,23 @@ public void executeQuery(
         Client client = neptuneConnection.getNeptuneClientConnection();
         GraphTraversalSource graphTraversalSource = neptuneConnection.getTraversalSource(client);
         GraphTraversal graphTraversal = null;
-        String labelName = recordsRequest.getTableName().getTableName();
+        String labelName;
+        String gremlinQuery = null;
+        if (recordsRequest.getConstraints().isQueryPassThrough()) {
+            Map<String, String> qptArguments = recordsRequest.getConstraints().getQueryPassthroughArguments();
+            queryPassthrough.verify(qptArguments);
+            labelName = qptArguments.get(NeptuneQueryPassthrough.COLLECTION);
+            gremlinQuery = qptArguments.get(NeptuneQueryPassthrough.QUERY);
+
+            ScriptEngine engine = new GremlinGroovyScriptEngine();
```

Comment:
here is duplicate code, same as `[NeptuneMetadataHandler.java](https://github.com/awslabs/aws-athena-query-federation/pull/1886/files#diff-34ceee35d3573bb98182e81a32b1a4840d80b442a1bdf59c59e800dd48413195)` .

Reply:
Hi @chngpe Thank you, removed duplicate code and made a common function.